repose-aggregator/components/filters/ip-user-filter/src/test/scala/org/openrepose/filters/ipuser/IpUserFilterTest.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+
+package org.openrepose.filters.ipuser
+
+import org.junit.runner.RunWith
+import org.openrepose.commons.utils.servlet.http.HttpServletRequestWrapper
+import org.openrepose.filters.ipuser.config.{GroupType, IpUserConfig}
+import org.scalatest.{BeforeAndAfter, FunSpec, Matchers}
+import org.scalatest.junit.JUnitRunner
+import org.springframework.mock.web.{MockFilterChain, MockHttpServletRequest, MockHttpServletResponse}
+
+@RunWith(classOf[JUnitRunner])
+class IpUserFilterTest extends FunSpec with BeforeAndAfter with Matchers {
+
+  val DefaultQuality = ";q=0.4"
+
+  var ipUserFilter: IpUserFilter = _
+  var servletRequest: MockHttpServletRequest = _
+  var servletResponse: MockHttpServletResponse = _
+  var filterChain: MockFilterChain = _
+
+  before {
+    ipUserFilter = new IpUserFilter(null)
+    servletRequest = new MockHttpServletRequest
+    servletResponse = new MockHttpServletResponse
+    filterChain = new MockFilterChain
+  }
+
+  describe("setting the request x-pp-user header") {
+    it("should use the value of the remote IP address of the request when no X-Forwarded-For header is included") {
+      ipUserFilter.configurationUpdated(createConfig())
+      servletRequest.setRemoteAddr("10.1.2.3")
```

Comment:
After a second look at the `BeforeAndAfter` trait, it's `before` method is run before each test. The nuance between the two is that a failure in the `before` method will abort the rest of the suite marking the remainder as skipped and a failure in the `beforeEach` method will continue to fail each test of the suite.

Maybe we should be using only the `BeforeAndAfterAll` and `BeforeAndAfterEach` traits instead of the `BeforeAndAfter` trait across the entire codebase. There was also a weird deprecation and then reuse of the `BeforeAndAfter` trait between ScalaTest 1.0 and 2.0. That lends even more to not using it, if only for the explicitness of the other two.

Thoughts?